numpy/lib/function_base.py

```diff
@@ -514,8 +514,7 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
-        wgt = np.asarray(weights)
+        wgt = np.asarray(weights, dtype=float)
```

Comment:
Maybe asarray needs an argument to say 'make the dtype at least this high
on the casting graph (higher is OK)'?
On 30 Jan 2015 16:50, "Julian Taylor" <notifications@github.com> wrote:

> In numpy/lib/function_base.py
> <https://github.com/numpy/numpy/pull/5525#discussion_r23854432>:
>
> > @@ -514,8 +514,7 @@ def average(a, axis=None, weights=None, returned=False):
> >          avg = a.mean(axis)
> >          scl = avg.dtype.type(a.size/avg.size)
> >      else:
> > -        a = a + 0.0
> > -        wgt = np.asarray(weights)
> > +        wgt = np.asarray(weights, dtype=float)
>
> unfortunately that won't work as result_type needs an ndarray as input,
> which is was asarray creates
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/5525/files#r23854432>.
>